# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/performance/PerformanceStorageWorker.cpp
# Commit: 9ee546b9fd2b
# Full Hash: 9ee546b9fd2b093fbb40ff17870fa70df287986f
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2022-02-26 21:36:36
# Regressor Bug: 1754365
# File Overlap Count: 1
# Description:
#   Bug 1754365 - Saving NavigationPreload's PerformanceTimingData into worker's PerformanceStorage. r=dom-worker-reviewers,jesup
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138252
# ==============================================================================

diff -r cf972fe0d961 -r 9ee546b9fd2b dom/performance/PerformanceStorageWorker.cpp
--- a/dom/performance/PerformanceStorageWorker.cpp	Sat Feb 26 08:40:26 2022 +0000
+++ b/dom/performance/PerformanceStorageWorker.cpp	Sat Feb 26 08:40:26 2022 +0000
@@ -120,6 +120,19 @@
   Unused << NS_WARN_IF(!r->Dispatch());
 }
 
+void PerformanceStorageWorker::AddEntry(
+    const nsString& aEntryName, const nsString& aInitiatorType,
+    UniquePtr<PerformanceTimingData>&& aData) {
+  if (!aData) {
+    return;
+  }
+
+  UniquePtr<PerformanceProxyData> data = MakeUnique<PerformanceProxyData>(
+      std::move(aData), aInitiatorType, aEntryName);
+
+  AddEntryOnWorker(std::move(data));
+}
+
 void PerformanceStorageWorker::ShutdownOnWorker() {
   MutexAutoLock lock(mMutex);
 